### DISCUSSION TITLE
Adding custom directive to resolver in schema

### DISCUSSION THREAD

**Discussion by simonvandel** [async-graphql/async-graphql#869] [Q&A]
Hi

I am trying to setup a schema so that the `schema.sdl()` will output something like
```gql
directive @MyDirective(key: String) on FIELD_DEFINITION
type Query {
   resolveA @MyDirective(key: "")
}
```

The docs mention how to create the custom directive `MyDirective` https://docs.rs/async-graphql/3.0.35/async_graphql/attr.Directive.html , but how do I annotate that `resolveA` should generate the `MyDirective` in the outputted SDL?

**Comment by sunli829** [comment]
This is not supported yet, and it may not be supported until much later. 🙂

**Reply by simonvandel** [reply]
Alright, thanks for the quick reply. So I guess the only "workaround" i can do currently is manually patching the String returned from `.sdl()` to contain the directive?

Would you accept a contribution that added support for this? In any case, I/someone else might need some pointers/mentoring on how best to add this feature.
